✨ feat: Implement API Key management functionality

MarioJames opened the pull request:
#### 💻 变更类型 | Change Type

<!-- For change type, change [ ] to [x]. -->

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] 📝 docs
- [ ] 🔨 chore

#### 🔀 变更说明 | Description of Change

在 LobeChat 中初始化 API-KEY 功能相关表结构及工具
本次提交包含几个方面的内容：

##### 数据访问层
- **Schema 定义** (`src/database/schemas/apiKey.ts`)：使用 Drizzle ORM 定义数据表结构
- **Model 层** (`src/database/models/apiKey.ts`)：封装数据库操作逻辑
- **迁移文件**：添加数据库表结构迁移

数据库约束
1. 主键约束: id 字段作为自增主键
2. 唯一约束: key 字段必须全局唯一
3. 外键约束: user_id 关联 users 表，支持级联删除
4. 非空约束: name, key, user_id 字段不能为空
5. 时间戳: 自动维护 created_at 和 updated_at 时间戳
表关联关系 ER 图如下，User 与 APIKey 关联关系为 1:n
![image](https://github.com/user-attachments/assets/1faad12f-d7d9-4275-bb20-c93cd0f2f3bf)


##### 服务层
- **接口定义** (`src/services/apiKey/type.ts`)：定义统一的服务接口
- **客户端服务** (`src/services/apiKey/client.ts`)：客户端模式下的服务实现
- **服务器端服务** (`src/services/apiKey/server.ts`)：服务器模式下的服务实现
- **自适应选择** (`src/services/apiKey/index.ts`)：根据运行环境自动选择服务实现

##### API 路由层
- **TRPC 路由** (`src/server/routers/lambda/apiKey.ts`)：提供类型安全的 API 接口，有两个比较特殊的接口
  -  `create` 接口：创建API-Key 时，对生成的字符串进行加密存储
  - `findByKey`：由于在数据库中使用密文存储，用户看到的是明文，所以在通过明文查找对应的 API-KEY 信息时，在生成查找条件时，需要进行一次明文-密文的转换
- **输入验证**：使用 Zod 进行严格的参数校验
- **权限控制**：集成用户认证中间件

##### 工具函数
- **密钥生成** (`src/utils/apiKey.ts`)：安全的随机密钥生成算法
- **格式验证**：严格的密钥格式校验
- **过期检查**：密钥有效性验证逻辑

#### 📝 补充信息 | Additional Information

页面路径：账户管理 - API Key 管理

## Summary by Sourcery

Implement end-to-end API Key management functionality including database schema, ORM model, service and API routes, feature-flagged UI components, and localization support.

New Features:
- Add database schema, migration and model for API Key management with encryption, validation, and expiration support
- Provide TRPC router with procedures to create, list, update, delete, fetch, and validate API keys
- Implement service layer switching between client and server modes for handling API Key operations
- Add full UI in the profile section including a new API Key tab, table view, editable cells, date picker, create modal, and key display component
- Introduce feature flag "api_key_manage" to control visibility of API Key management functionality

Enhancements:
- Extend rbac_roles schema with a new JSONB metadata column

Documentation:
- Add i18n translations for all new API Key management UI strings and tab label

MarioJames committed:
✨ feat: Implement API Key management functionality

- Added new components for API Key management including creation, deletion, and display.
- Introduced a new database schema for storing API Keys.
- Implemented server and client services for API Key operations.
- Integrated API Key management into the profile section with appropriate routing and feature flags.
- Enhanced localization support for API Key related UI elements.

This commit lays the groundwork for managing API Keys within the application, allowing users to create, view, and manage their keys effectively.

MarioJames committed:
fix: server config unit test

MarioJames committed:
Merge branch 'main' into sprint/apikey

arvinxx commented on src/database/migrations/0029_add_apikey_manage.sql:
```diff
@@ -0,0 +1,16 @@
+CREATE TABLE "api_keys" (
```
这里防御性一下，`CREATE TABLE IF NOT EXISTS`

MarioJames committed:
Merge branch 'main' into sprint/apikey

MarioJames committed:
✨ feat(database): Create api_keys table with conditional existence check

- Added a conditional check to create the "api_keys" table only if it does not already exist.
- Ensured the foreign key constraint for "user_id" references the "users" table remains intact.

This change enhances the migration process by preventing errors during table creation if the table already exists.

MarioJames committed:
Merge branch 'main' into sprint/apikey

arvinxx commented on src/services/apiKey/client.ts:
我感觉是不是不需要有 client 的 api key ？只需要一个 server 的api key 就好了

MarioJames committed:
Merge branch 'main' into sprint/apikey

MarioJames committed:
feat: Implement API Key management interface

- Introduced a new Client component for managing API keys, including creation, updating, and deletion functionalities.
- Replaced the previous page component with the new Client component in the API key management page.
- Removed obsolete client and server service files related to API key management, streamlining the service layer.

This update enhances the user experience by providing a dedicated interface for API key operations.
